pkg/innerring/container.go

```diff
@@ -0,0 +1,61 @@
+package innerring
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+
+	neofscrypto "github.com/nspcc-dev/neofs-sdk-go/crypto"
+	neofsecdsa "github.com/nspcc-dev/neofs-sdk-go/crypto/ecdsa"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+)
+
+// authSystem encapsulates NeoFSID contract's interface of the NeoFS Sidechain
+// and provides interface needed by the Inner Ring application.
+//
+// Implements container.AuthSystem.
+type authSystem struct {
+	neofsID neoFSIDContract
+}
+
+// init initializes the authSystem instance.
+func (x *authSystem) init(neofsID neoFSIDContract) {
```

Comment:
I don't really like initializing instances by filling the fields since such an approach reveals the internal data (yeah, this is the same package, but still). So, it should be a `new___` constructor or a method. I prefer method since it allows to init the instance when it is an internal field of smth.

Reply:
Ok, just affraid that it'll have more fields in the future => it'll require `prm` struct => it'll require creating already two vars. And everything is for an ordinary private helper struct only.

Reply:
Normally any component shouldn't depent on >3-4 entities, otherwise it becomes too heavy.